System/Console/ANSI/Color.hs

```diff
@@ -2,50 +2,59 @@
 module System.Console.ANSI.Color
   (
   -- | 8-bit to 24-bit color conversion
-    color8To24
+    xterm256ToSRGB
   -- | Utility functions
   , clamp
-  , color8CodeToEither
   , closest4bitsANSIColor
+  , closestXterm256Color
+  , color8CodeToXterm256
+  , colorToCode
+  , xterm256ColorToCode
     ) where
 
 
-import qualified Data.Colour.SRGB.Linear as SRGBLin(rgb, Colour)
-import qualified Data.Colour.SRGB as SRGB (RGB (..), toSRGB, Colour)
+import Data.Colour.SRGB (Colour, RGB (..), sRGB, toSRGB)
 
 import Data.Word (Word8)
 import Data.Colour.Names (black, blue, cyan, green, grey, lime, magenta, maroon,
                           navy, olive, purple, red, silver, teal, white, yellow)
 
 import Data.List (minimumBy, (!!))
+import Data.Maybe (fromJust)
 
 import System.Console.ANSI.Types
 
--- | Returns a color, in the linear sRGB colorspace, whose components are the components
---  of an 8-bit ANSI color.
+-- | Returns a color, in the linear sRGB colorspace, whose components are the
+-- components of an 8-bit ANSI color.
 -- For each color component, we first clamp the value to its admissible range
 -- defined in https://en.wikipedia.org/wiki/ANSI_escape_code#Colors:
 --  - 0-5 for 8-bit rgb
 --  - 0-23 for 8-bit grayscale
 -- Then we apply the xterm mapping (the mapped value is in range [0..255]).
 -- Finally we interpret the xterm-mapped values as components of a color
--- in linear srgb colorspace (we divide by 255 to express the value as a floating point in range [0 1]).
-color8To24 :: Color8 -> SRGBLin.Colour Float
-color8To24 (RGB8Color r' g' b') =
-  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapRGB8bitComponent . (\v -> clamp v 0 5) :: Word8 -> Float
-      r = mapping r'
-      g = mapping g'
-      b = mapping b'
-  in  SRGBLin.rgb r g b
-color8To24 (Gray8Color g') =
-  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapGray8bitComponent . (\v -> clamp v 0 23) :: Word8 -> Float
-      g = mapping g'
-  in  SRGBLin.rgb g g g
+-- in linear srgb colorspace (we divide by 255 to express the value as a
+-- floating point in range [0 1]).
+xterm256ToSRGB :: Xterm256Color -> Colour Float
+xterm256ToSRGB (SystemColor intensity color)
+  = fromJust $ lookup (intensity, color) aNSIColors
+xterm256ToSRGB (RGBColor (RGB r' g' b')) = sRGB r g b
```

Comment:
So you chose to change the colorspace here? (from linear srgb to srgb). I must say I'm not sure of my initial choice, I know from reading the implementations that it's faster because there is no power function involved, but I don't know if it's correct. I think it would require a bit of investigation and documentation but that can be probably done another time. Or do you already know why you chose srgb here?

Reply:
When I implemented `SetRGBColor` for `SGR`, I think I was influenced by (1) the `colour` package's objective of being the standard colour library for Haskell and (2) `Data.Colour.Names` being created by `sRGB24` (and, hence, being in the sRGB colour space). I was not optimizing for speed - partly because I was conscious that I/O is inherently 'slow' - but looking for a 'default' representation of colour.